### ISSUE TITLE
Optimization regression

### ISSUE THREAD

**Issue by LeventErkok** [Z3Prover/z3#6643]
This is a recent breakage; so hopefully will be easy to locate.

For this benchmark:

```
(set-logic ALL)
(declare-datatypes ((T 2)) ((par (T1 T2) ((mkT (fst T1) (snd T2))))))

(declare-fun t () (T Int Int))
(define-fun  x () Int (fst t))
(define-fun  y () Int (snd t))

(assert (and (>= x 0) (<= x 100)))
(assert (and (>= y 0) (<= y 100)))
(assert (< (- x y) 10))
(assert (= (+ x y) 80))

(maximize x)
(maximize y)
(check-sat)
(get-objectives)
```

I'm getting the following output from z3 built this morning, March 22, 2023:

```
sat
(objectives
 ((fst t)  (interval 40 49))
 ((snd t) 40)
)
```

This output is incorrect, there shouldn't be an interval on `fst t`. 

z3 built on March 6th, 2023 produces the correct output:

```
sat
(objectives
 ((fst t) 44)
 ((snd t) 36)
)
```

**Comment by NikolajBjorner** [comment]
C:\z3\build>z3 6643.smt2
ASSERTION VIOLATION
File: C:\z3\src\math/lp/lp_primal_core_solver_tableau_def.h
Line: 311
this->column_is_feasible(entering)

**Comment by NikolajBjorner** [comment]
@levnach - likely easy, but if you have bandwidth to check...

**Comment by zhendongsu** [comment]
Another (much) simpler related instance:
```
[519] % z3debug small.smt2
ASSERTION VIOLATION
File: ../src/math/lp/lp_primal_core_solver_tableau_def.h
Line: 311
this->column_is_feasible(entering)
(C)ontinue, (A)bort, (S)top, (T)hrow exception, Invoke (G)DB
a
[520] % cat small.smt2
(declare-fun a () Int)
(declare-fun b () Int)
(minimize (* a b))
(check-sat)
```

**Comment by NikolajBjorner** [comment]
fixed by 130400d76e3cc37e4b6f3a9ed6e8a4781b7f3599
